src/unity/python/turicreate/toolkits/_pre_trained_models.py

```diff
@@ -15,14 +15,24 @@
 
 MODELS_URL_ROOT = 'https://docs-assets.developer.apple.com/turicreate/models/'
 
+
 def _get_model_cache_dir():
-    tmp_dir = _tc.config.get_runtime_config()['TURI_CACHE_FILE_LOCATIONS']
-    return _os.path.join(tmp_dir, 'model_cache')
+    cache_dir = _tc.config.get_runtime_config()['TURI_CACHE_FILE_LOCATIONS']
+    download_path = _os.path.join(cache_dir, 'model_cache')
 
-def _download_and_checksum_files(urls, dirname, delete=False):
-    if not _os.path.exists(dirname):
-        _os.makedirs(dirname)
+    if not _os.path.exists(download_path):
+        try:
+            _os.makedirs(download_path)
+        except:
+            raise RuntimeError("Could not write to the turicreate file cache, which is currently set to \"{cache_dir}\".\n"
+                               "To continue you must update this location to a writable path by calling:\n"
+                               "\ttc.config.set_runtime_config(\'TURI_CACHE_FILE_LOCATIONS\', <path>)\n"
+                               "Where <path> is a writable file path that exists.".format(cache_dir=cache_dir))
```

Comment:
It's also possible to set the env var with this name too. Is this worth mentioning briefly?

Reply:
Also "... that exists" is technically not true, since we attempt to create all the paths. As long we have permission to do so. It's a subtle distinction that is not that important though, and I agree that it's better for users that hit this issue to manually confirm that there is a directory there.

Reply:
Thanks for the review. 

I don't believe it's worth mentioning the environment variable. This error message is already long enough.

Surprisingly the "that exists" part is accurate. See #789 which I discovered while fixing this bug.